lighthouse-core/lib/proto-preprocessor.js

```diff
@@ -25,17 +25,6 @@ function processForProto(result, opts = {}) {
   /** @type {LH.Result} */
   const reportJson = JSON.parse(result);
 
-  // Clean up the configSettings
-  // Note: This is not strictly required for conversion if protobuf parsing is set to
-  // 'ignore unknown fields' in the language of conversion.
-  if (reportJson.configSettings) {
```

Comment:
why is this being removed?

Reply:
It was pruning the JSON sent back to LR, which would then place it in a Config proto that only had `{emulatedFormFactor, locale, onlyCategories}`. Since the rest weren't saved it made sense to cut the extras here. With the proto changes, most of the settings (i left out things like `output`) can be saved, so I'm undoing the pruning. The fields that aren't in the proto are safely ignored when serialized.

Reply:
Gotcha, yeah basically I'm wondering why we don't need to prune anymore even though we still don't support 100% of the settings :)

Are the remaining extra properties any less of a concern than it was before? Based on the comments here, it seems like it was still technically unnecessary then but we decided to do it anyway 🤷‍♂️

Reply:
it's just a couple string values now. before it was like, 95% of it. pruning is slightly more unnecessary now :)